File: app/schemas/me.py
"""Schemas para /api/v1/me/*."""
from __future__ import annotations

from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel


SemaphoreStatus = Literal["green", "yellow", "red", "pending"]
LoadStatus = Literal["green", "yellow", "red"]


class MechanicSummary(BaseModel):
    id: str
    level: str
    current_load_hrs: float
    available_hrs: float
    load_status: LoadStatus


class VehicleBrief(BaseModel):
    plates: Optional[str] = None
    brand: Optional[str] = None
    model: Optional[str] = None


class WorkOrderBrief(BaseModel):
    id: str
    order_number: str
    type: str
    priority: str
    vehicle: VehicleBrief


class LineBrief(BaseModel):
    id: str
    service_name: str
    service_required_level: str
    standard_duration_hrs: Optional[float] = None
    status: str
    bay_name: Optional[str] = None


class TimerState(BaseModel):
    started_at: Optional[datetime] = None
    elapsed_minutes: int
    remaining_estimated_minutes: Optional[int] = None
    semaphore: SemaphoreStatus


class PartsStatus(BaseModel):
    total: int
    available: int
    blocking: bool


class MyTaskItem(BaseModel):
    assignment_id: str
    work_order: WorkOrderBrief
    line: LineBrief
    timer: TimerState
    parts_needed: PartsStatus
    available_actions: list[str]


class MyTasksResponse(BaseModel):
    mechanic: MechanicSummary
    items: list[MyTaskItem]
    summary: dict


class FindingReportRequest(BaseModel):
    description: str
    suggested_service_id: Optional[str] = None
    estimated_extra_hrs: Optional[float] = None

File: frontend/src/api/endpoints/me.ts
/** Client de /api/v1/me/*. */
import api from "@/api/client";

export interface MyTaskItem {
  assignment_id: string;
  work_order: {
    id: string;
    order_number: string;
    type: string;
    priority: string;
    vehicle: { plates: string | null; brand: string | null; model: string | null };
  };
  line: {
    id: string;
    service_name: string;
    service_required_level: string;
    standard_duration_hrs: number | null;
    status: string;
    bay_name: string | null;
  };
  timer: {
    started_at: string | null;
    elapsed_minutes: number;
    remaining_estimated_minutes: number | null;
    semaphore: "green" | "yellow" | "red" | "pending";
  };
  parts_needed: { total: number; available: number; blocking: boolean };
  available_actions: string[];
}

export interface MyTasksResponse {
  mechanic: {
    id: string;
    level: string;
    current_load_hrs: number;
    available_hrs: number;
    load_status: "green" | "yellow" | "red";
  };
  items: MyTaskItem[];
  summary: Record<string, number>;
}

export interface FindingReportPayload {
  description: string;
  suggested_service_id?: string;
  estimated_extra_hrs?: number;
}

export interface FindingRead {
  id: string;
  work_order_id: string;
  work_order_line_id: string | null;
  description: string;
  suggested_service_id: string | null;
  estimated_extra_hrs: number | null;
  status: "pending" | "approved" | "rejected";
  reported_by: string | null;
  reviewed_by: string | null;
  rejection_reason: string | null;
  resulting_line_id: string | null;
  created_at: string;
}

export const meApi = {
  getTasks: () => api.get<MyTasksResponse>(`/v1/me/tasks`).then((r) => r.data),

  reportFinding: (lineId: string, payload: FindingReportPayload) =>
    api.post<FindingRead>(`/v1/me/tasks/${lineId}/findings`, payload).then((r) => r.data),
};
